src/main/java/updater/DownloadProgressTracker.java

```diff
@@ -0,0 +1,40 @@
+package updater;
+
+import javafx.application.Platform;
+import javafx.scene.control.ProgressBar;
+
+import java.net.URL;
+import java.util.Optional;
+
+/**
+ * This class is used to track download if a UI element (i.e. progress bar) is needed to show the download progress
+ */
+public class DownloadProgressTracker {
+    private final String downloadLabel;
+    private final URL urlIdentifier;
+    private Optional<ProgressBar> progressBar;
```

Comment:
Could we pass the progress bar in on construction? It's a bit awkward to construct these objects piecemeal.

If we want to be more OO, this is really a progress bar augmented with a label and url. Maybe we could name it something which reflects that?